plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/impl/FlightStreamManager.java

```diff
@@ -0,0 +1,205 @@
+import org.opensearch.arrow.spi.StreamProducer;
+import org.opensearch.arrow.spi.StreamReader;
+import org.opensearch.arrow.spi.StreamTicket;
+import org.opensearch.arrow.spi.StreamTicketFactory;
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+/**
+ * FlightStreamManager is a concrete implementation of StreamManager that provides
+ * an abstraction layer for managing Arrow Flight streams in OpenSearch.
+ * It encapsulates the details of Flight client operations, allowing consumers to
+ * work with streams without direct exposure to Flight internals.
+ */
+public class FlightStreamManager implements StreamManager {
+    private static final Logger logger = LogManager.getLogger(FlightStreamManager.class);
+
+    private FlightStreamTicketFactory ticketFactory;
+    private FlightClientManager clientManager;
+    private Supplier<BufferAllocator> allocatorSupplier;
+    private final Cache<String, StreamProducerHolder> streamProducers;
+    // TODO read from setting
+    private static final TimeValue DEFAULT_CACHE_EXPIRE = TimeValue.timeValueMinutes(10);
+    private static final int MAX_WEIGHT = 1000;
+
+    /**
+     * Holds a StreamProducer along with its metadata and resources
+     */
+    record StreamProducerHolder(StreamProducer<VectorSchemaRoot, BufferAllocator> producer, BufferAllocator allocator, long creationTime,
+        AtomicReference<VectorSchemaRoot> root) {
+        public StreamProducerHolder {
+            Objects.requireNonNull(producer, "StreamProducer cannot be null");
+            Objects.requireNonNull(allocator, "BufferAllocator cannot be null");
+        }
+
+        static StreamProducerHolder create(StreamProducer<VectorSchemaRoot, BufferAllocator> producer, BufferAllocator allocator) {
+            return new StreamProducerHolder(producer, allocator, System.nanoTime(), new AtomicReference<>(null));
+        }
+
+        boolean isExpired() {
+            return System.nanoTime() - creationTime > producer.getJobDeadline().getNanos();
+        }
+
+        /**
+         * Gets the VectorSchemaRoot associated with the StreamProducer.
+         * If the root is not set, it creates a new one using the provided BufferAllocator.
+         */
+        public VectorSchemaRoot getRoot() {
+            return root.updateAndGet(current -> current != null ? current : producer.createRoot(allocator));
+        }
+    }
+
+    /**
+     * Constructs a new FlightStreamManager.
+     */
+    public FlightStreamManager() {
+        this.streamProducers = CacheBuilder.<String, StreamProducerHolder>builder()
+            .setExpireAfterWrite(DEFAULT_CACHE_EXPIRE)
+            .setMaximumWeight(MAX_WEIGHT)
```

Comment:
Let's not to forget to properly close the producer on eviction:

```
.removalListener(n -> {
                if (n.getRemovalReason() != RemovalReason.EXPLICIT) {
                    try (var unused = n.getValue().producer()) {
                    } catch (IOException e) {
                        logger.error("Error closing stream producer, this may cause memory leaks.", e);
                    }
                }
            })
```